### PR TITLE
Added feedback form and sitemap to the footer

### PR DESCRIPTION
**Pull request by RajatX24**
I have added the feedback form and the sitemap to the footer as requested in issue.

This is the sample image from the issue, showing how the footer should look:
sample

BEFORE MAKING CHANGES:
before

AFTER MAKING CHANGES:
after1

MOBILE VIEW:
after2

NOTE: In order to link the form in footer to your formspree form
1) go to formspree
2)register and create a form
3)copy the form code of your form
4)create a .env file in frontend folder
5)inside it write as follows: REACT_APP_FORMSPREE_FORM_FORM_CODE =  your-form-code-here

After doing the above the form should work.

I request the maintainer to please mark this as 'hacktoberfest-accepted' while merging this. I want this contribution to count for hacktober fest.

Thanks!

## Summary by CodeRabbit

- **New Features**
	- Introduced a responsive `Footer` component with a feedback form for user submissions.
	- Added a `ResponseSuccess` component to display confirmation messages upon successful feedback submission.
	- Integrated the `Footer` component into the `Home` page layout.
- **Configuration Updates**
	- Added new environment variables for forms, GitHub repository URL, and contact email.
- **Dependencies**
	- Added new dependencies: `@formspree/react` and `dotenv`.
	- Introduced `dotenv-webpack` as a development dependency.

### CODE REVIEW

**Comment by RajatX24** [comment]
Please let me know if there are any changes that need to be made.

**Comment by shravan20** [comment]
> Please let me know if there are any changes that need to be made.

Can you address the PR review comments by the coderabbitai?

**Comment by RajatX24** [comment]
# made changes suggested by coderabbitai


## 👨‍💻 Changes made
- Used Material-UI's styling solution instead of inline styles.
- Used function declaration and added PropTypes.
- Improved sitemap maintainability by using constants or environment variables for URLs
- Improved the feedback form by removing redundant success message and disabling the submit button while the form is being submitted

## Changes not made
- Consider using a translation function for the success message. 
   I did not used translation functions as i feel its beyond the scope of this PR. The project does not use tranlation functions as of now. Hence i haven't used translation functions despite coderabbitai having mentioned it.
- Footer component is missing the sitemap element.
 This was a wrong analysis by coderabbitai. The footer component does have the sitemap. It can clearly be seen in the screenshots attached above.

## Note to reviewers
If there are any changes that need to be made please let me know.
If merging please add the Hacktoberfest label. I would be grateful.


## 📷 Screenshots

### ADDED MUI'S STYLING SOLUTION INSTEAD OF INLINE STYLES:
styles

### ADDED PROP TYPES:
prop

### USED CONSTANTS AND ENVIRONMENT VARIABLES FOR SITEMAP MAINTAINABILITY
var1
var2

File: frontend/.env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_FORMSPREE_FORM_CODE = abcdef
+REACT_APP_GITHUB_REPO_URL = https://github.com/shravan20/github-readme-quotes
+REACT_APP_CONTACT_EMAIL = your_contact_email@example.com
```
Comment (shravan20) [inline frontend/.env.example:3]:
```suggestion
REACT_APP_CONTACT_EMAIL = mrshravankumarb@gmail.com
```

File: frontend/.env.example [hunk 2]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_FORMSPREE_FORM_CODE = abcdef
```
Comment (shravan20) [inline frontend/.env.example:1]:
```suggestion
REACT_APP_FORMSPREE_FORM_CODE = moqgpayj
```

File: frontend/.env.example (hunk 1 above)
Comment (shravan20) [inline frontend/.env.example:3]:
```suggestion
REACT_APP_CONTACT_EMAIL = your_contact_email@example.com

```

File: frontend/.env.example [hunk 3]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_FORMSPREE_FORM_CODE = moqgpayj
+REACT_APP_GITHUB_REPO_URL = https://github.com/shravan20/github-readme-quotes
+REACT_APP_CONTACT_EMAIL = mrshravankumarb@gmail.com
```
Comment (shravan20) [inline frontend/.env.example:3]:
```suggestion
REACT_APP_CONTACT_EMAIL = mrshravankumarb@gmail.com

```

File: frontend/src/components/Footer/Footer.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    <Link href={`${GITHUB_REPO_URL}/blob/main/LICENSE`} display="block" style={{ marginBottom: '10px' }} color="textSecondary">
+                        Terms and Conditions
+                    </Link>
+                </Paper>
+            </Grid>
+        </Grid>
+    )
+}
+
+export default Footer;
```
Comment (shravan20) [inline frontend/src/components/Footer/Footer.js:72]:
```suggestion
export default Footer;

```

File: frontend/src/components/Footer/ResponseSuccess.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+function ResponseSuccess(){
+    const classes = useStyles();
+    return (<Paper className={classes.paper}>
+        <Typography variant="h6" color="primary" gutterBottom>Your response has been recorded successfully!</Typography>
+    </Paper>)
+}
+
+ResponseSuccess.propTypes = {};
+
+export default ResponseSuccess;
```
Comment (shravan20) [inline frontend/src/components/Footer/ResponseSuccess.js:26]:
```suggestion
export default ResponseSuccess;

```

**Pull request merged**